ui/src/status/components/NewsFeed.js

```diff
@@ -0,0 +1,79 @@
+import React, {Component, PropTypes} from 'react'
+import {connect} from 'react-redux'
+import {bindActionCreators} from 'redux'
+
+import {fetchJSONFeedAsync} from 'src/status/actions'
+
+import FancyScrollbar from 'shared/components/FancyScrollbar'
+import JSONFeedReader from 'src/status/components/JSONFeedReader'
+
+class NewsFeed extends Component {
+  constructor(props) {
+    super(props)
+  }
+
+  // TODO: implement shouldComponentUpdate based on fetching conditions
+
+  render() {
+    const {hasCompletedFetchOnce, isFetching, isFailed, data} = this.props
+
+    if (!hasCompletedFetchOnce) {
+      return isFailed
+        ? <span>Failed to load NewsFeed.</span>
+        : // TODO: Factor this out of here and AutoRefresh
+          <div className="graph-fetching">
+            <div className="graph-spinner" />
+          </div>
+    }
+
+    return (
+      <FancyScrollbar autoHide={false} className="newsfeed--container">
+        {isFetching
+          ? // TODO: Factor this out of here and AutoRefresh
+            <div className="graph-panel__refreshing">
+              <div />
+              <div />
+              <div />
+            </div>
+          : null}
+        {isFailed ? <span>Failed to refresh NewsFeed</span> : null}
+        <JSONFeedReader data={data} />
+      </FancyScrollbar>
+    )
+  }
```

Comment:
I don't think we should support multiple failure states and multiple refresh states. In the future, let's consolidate into:

```js
if (isFetching) {
} else if (isFailed) {
} else { // Not fetching, not failed, so successful fetch was performed
}